pkg/client/query/cache/config.go

```diff
@@ -13,57 +13,86 @@ const (
 	LeastFrequentlyUsed
 )
 
-// queryCacheConfig is the configuration for query caches. It is intended to be
-// configured via QueryCacheOptionFn functions.
+// queryCacheConfig is the configuration for query caches.
+// It is intended to be configured via QueryCacheOptionFn functions.
 type queryCacheConfig struct {
-	// MaxKeys is the maximum number of items (key/value pairs) the cache can
+	// maxKeys is the maximum number of key/value pairs the cache can
 	// hold before it starts evicting.
-	MaxKeys        int64
-	EvictionPolicy EvictionPolicy
-	// TTL is how long items should remain in the cache. Items older than the TTL
-	// MAY not be evicted but SHOULD not be considered as cache hits.
-	TTL time.Duration
+	maxKeys int64
 
-	// historical determines whether the cache will cache a single value for each
-	// key (false), or whether it will cache a history of values for each key (true).
+	// TODO_CONSIDERATION:
+	//
+	// maxValueSize is the maximum cumulative size of all values in the cache.
+	// maxValueSize int64
+	// maxCacheSize is the maximum cumulative size of all keys AND values in the cache.
+	// maxCacheSize int64
+
+	// evictionPolicy determines which values are removed when number of keys in the cache reaches maxKeys.
+	evictionPolicy EvictionPolicy
+	// ttl is how long values should remain valid in the cache. Items older than the
+	// ttl MAY NOT be evicted immediately, but are NEVER considered as cache hits.
+	ttl time.Duration
+	// historical determines whether each key will point to a single values (false)
+	// or a history (i.e. reverse chronological list) of values (true).
 	historical bool
-	// pruneOlderThan is the number of past blocks for which to keep historical
-	// values. If 0, no historical pruning is performed. It only applies when
-	// historical is true.
-	pruneOlderThan int64
+	// maxVersionAge is the max difference between the latest known version and
```

Comment:
this sentence is hard to grok.

Any chance you could improve it with an example?